### PR TITLE
Add basic numeric calc and bool type

### PR DESCRIPTION
**Pull request by GreasySlug**
Related #178.

Four arithmetic operations, comparisons, and logical operations were missing, so those were added in Japanese.

Once this has been reviewed, then I' ll translate it into English and add it later.

@mtshiba

### CODE REVIEW

File: doc/JA/syntax/00_basic.md [hunk 1]
```diff
@@ -34,10 +34,111 @@ print!(f x, y) # NG, `print!(f(x), y)`または`print!(f(x, y))`の二通りの
...
+111111
+>>> (21 * 37) % 7
+0
+```
+
+ergには厳密な型があります。しかし、型推論が行われるため、型が明示的に宣言されていなければ勝手に型が推論されます。
+
+そのため、数値型であれば異なる型同士で計算をすることもできます。
+
+Rustなどの厳密な型のある言語とは違い暗黙的に型が変換されるので気をつけましょう。
```
Comment (GreasySlug) [inline doc/JA/syntax/00_basic.md:62]: I think the explanation here is different, so any better explanation would be appreciated.

**Comment by mtshiba** [comment]
I think it is unnecessary to bother writing about the arithmetic operations in the syntax documentation.

It is better to write in the documentation under the API what specific operators are defined.

File: doc/JA/syntax/00_basic.md (hunk 1 above)
Comment (mtshiba) [reply doc/JA/syntax/00_basic.md:62]: Erg’s type conversion is also strict, but has the flexibility of subtyping by trait or class.

Comment (mtshiba) [reply doc/JA/syntax/00_basic.md:62]: For example, given the operation `1.0 + 1`, the 1 on the right-hand side is cast to instance `1.0` in the superclass Float; the addition of Float and Float is defined, but the addition of Int and Float is not directly defined.

**Comment by GreasySlug** [comment]
> It is better to write in the documentation under the API what specific operators are defined.

Well, that may be true, but  four arithmetic operations and comparisons are often used in "quick tour" and "getting started.

like [python](https://docs.python.org/3/tutorial/), [ruby](https://www.ruby-lang.org/en/documentation/quickstart/)

> Erg’s type conversion is also strict, but has the flexibility of subtyping by trait or class.

This is exactly the kind of thing that should be explained in detail in the API.

> For example, given the operation 1.0 + 1, the 1 on the right-hand side is cast to instance 1.0 in the superclass Float; the addition of Float and Float is defined, but the addition of Int and Float is not directly defined.

so the examples are incorrect and I will fix them

**Comment by mtshiba** [comment]
Quick tour doc exists: https://github.com/erg-lang/erg/blob/main/doc/EN/syntax/quick_tour.md

**Comment by mtshiba** [comment]
True and False are singletons of type Bool, respectively.
Casting them to Int yields 1, 0, but they are not defined as 1, 0.
Bool operations are performed with `and`, `or`, and `not`. `&&`, `||`, and `~` are bitwise operators.

**Comment by GreasySlug** [comment]
Quick Tour link added in the Basics section.
I removed the basic four arithmetic operations and comparisons because it is written not to write them if the contents are duplicated in python.
However, compile errors due to type are left untouched.

**Comment by mtshiba** [comment]
Since `Bool` is a subtype of `Int` and `Int` is a subtype of `Ratio` (although the current implementation is `Float`), `True` can be cast to `1.0`.

However, `Float` cannot be compared by ==, so a compile error will occur.

スクリーンショット 2022-09-29 151103

**Comment by mtshiba** [comment]
> The reason for this is that Float does not satisfy the transitive law, which is the condition for equivalence relations.

**Comment by mtshiba** [comment]
> `Float` cannot be compared by ==, so a compile error will occur.

The reason for this is that Float does not satisfy the transitive law, which is the condition for equivalence relations.

**Comment by GreasySlug** [comment]
Thank you for reviewing

**Comment by mtshiba** [comment]
Thanks!

**Pull request merged**